server/python/src/routers/movies.py

```diff
@@ -0,0 +1,188 @@
+from fastapi import APIRouter, HTTPException, Query
+from src.database.mongo_client import db, get_collection
+from src.models.models import CreateMovieRequest, Movie, MovieFilter, SuccessResponse
+from typing import List
+from datetime import datetime
+from src.utils.errorHandler import create_success_response, create_error_response
+
+'''
+This file contains all the business logic for movie operations.
+Each method demonstrates different MongoDB operations using the PyMongo driver.
+
+Implemented Endpoints:
+- GET /api/movies/ : Retrieve a list of movies with optional filtering, sorting,
+    and pagination.
+- POST /api/movies/batch : Create multiple movies in a single request.
+
+
+'''
+router = APIRouter()
+#------------------------------------
+# Place get_movie_by_id endpoint here
+#------------------------------------
+
+"""
+    GET /api/movies/
+
+    Retrieve a list of movies with optional filtering, sorting, and pagination.
+
+    Query Parameters:
+        q (str, optional): Text search query (searches title, plot, fullplot).
+        genre (str, optional): Filter by genre.
+        year (int, optional): Filter by year.
+        min_rating (float, optional): Minimum IMDB rating.
+        max_rating (float, optional): Maximum IMDB rating.
+        limitNum (int, optional): Number of results to return (default: 20, max: 100).
+        skipNum (int, optional): Number of documents to skip for pagination (default: 0).
+        sortBy (str, optional): Field to sort by (default: "title").
+        sort_order (str, optional): Sort direction, "asc" or "desc" (default: "asc").
+
+    Returns:
+        SuccessResponse[List[Movie]]: A response object containing the list of movies and metadata.
+"""
+
+@router.get("/", response_model=SuccessResponse[List[Movie]])
+# Validate the query parameters using FastAPI's Query functionality.
+async def get_all_movies(
+    q:str = Query(default=None),
+    genre:str = Query(default=None),
+    year:int = Query(default=None),
+    min_rating:float = Query(default=None),
+    max_rating:float = Query(default=None),
+    limit_num:int = Query(default=20, ge=1, le=100),
+    skip_num:int = Query(default=0, ge=0),
+    sort_by:str = Query(default="title"),
+    sort_order:str = Query(default="asc")
+):
+
+    # This variable naming might not be ideal as Python has a function named filter, but it helps illustrate the point.
+    filter = {}
```

Comment:
```suggestion
    # This variable naming might not be ideal as Python has a function named filter, but it helps illustrate the point.
    filter_dict = {}
```
yeah, I don't love the shadowing. can we adjust it slightly?

Reply:
Inserted!